### PR TITLE
Bug fix/pass on error messages

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3953/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3966/

File: arangod/StorageEngine/TransactionState.cpp [hunk 1]
```diff
@@ -125,32 +125,39 @@ int TransactionState::addCollection(TRI_voc_cid_t cid, std::string const& cname,
...
-    return trxCollection->updateUsage(accessType, nestingLevel);
+    return res.reset(trxCollection->updateUsage(accessType, nestingLevel));
   }
 
   // collection not found.
 
   if (nestingLevel > 0 && AccessMode::isWriteOrExclusive(accessType)) {
     // trying to write access a collection in an embedded transaction
-    return TRI_ERROR_TRANSACTION_UNREGISTERED_COLLECTION;
+    return Result(TRI_ERROR_TRANSACTION_UNREGISTERED_COLLECTION,
```
Comment (graetzer) [inline arangod/StorageEngine/TransactionState.cpp]: can the compiler do RVO if you already return a named object above?

File: arangod/StorageEngine/TransactionState.cpp [hunk 2]
```diff
@@ -125,32 +125,39 @@ int TransactionState::addCollection(TRI_voc_cid_t cid, std::string const& cname,
...
-    return TRI_ERROR_TRANSACTION_UNREGISTERED_COLLECTION;
+    return Result(TRI_ERROR_TRANSACTION_UNREGISTERED_COLLECTION, 
+                  std::string(TRI_errno_string(TRI_ERROR_TRANSACTION_UNREGISTERED_COLLECTION)) + ": " + cname + 
+                  " [" + AccessMode::typeString(accessType) + "]");
   }
 
   if (!AccessMode::isWriteOrExclusive(accessType) &&
       (isRunning() && !_options.allowImplicitCollections)) {
-    return TRI_ERROR_TRANSACTION_UNREGISTERED_COLLECTION;
+    return Result(TRI_ERROR_TRANSACTION_UNREGISTERED_COLLECTION,
```
Comment (graetzer) [inline arangod/StorageEngine/TransactionState.cpp]: can the compiler do RVO if you already return a named object above?

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3976/

**Pull request merged**